File: vigenere.java
import java.util.Scanner;
import java.io.File;
import java.io.IOException;

public class vigenere {
    private static String readFile(String pathname) throws IOException {
        File file = new File(pathname);
        StringBuilder fileContents = new StringBuilder((int)file.length());

        try (Scanner scanner = new Scanner(file)) {
            while (scanner.hasNextLine()) {
                fileContents.append(scanner.nextLine());
            }

            return fileContents.toString().replaceAll("[^A-Za-z]+", "").toLowerCase();
        }
    }

    public static void printToConsole(String sFile, int limit) {
        for (int i = 0; i < sFile.length(); i++) {
            System.out.print(sFile.charAt(i));

            if ((i + 1) % limit == 0)
                System.out.println();
        }
        System.out.println();
    }

    public static String makingPlaintext(String plaintextArg) throws IOException {
        StringBuilder plaintextStr = new StringBuilder(readFile(plaintextArg));

        while (plaintextStr.length() < 512)
            plaintextStr.append('x');

        if (plaintextStr.length() > 512)
            plaintextStr.setLength(512);

        return plaintextStr.toString();
    }

    public static String modifyKeyString(String keyString) {
        StringBuilder keyStringSB = new StringBuilder(keyString);

        int initialKeyLen = keyStringSB.length();
        int keyTracker = 0;

        while (keyStringSB.length() < 512) {
            // Notice the keyTracker post-increment
            keyStringSB.append(keyStringSB.charAt(keyTracker++));

            if (keyTracker >= initialKeyLen)
                keyTracker = 0;
        }

        return keyStringSB.toString();
    }

    public static String buildingCiphertext(String keyString, String plaintextStr) {
        StringBuilder cipherStringSB = new StringBuilder();

        for (int i = 0; i < plaintextStr.length(); i++) {
            int plainTextDec = plaintextStr.charAt(i) - 'a';
            int keyDec = keyString.charAt(i) - 'a';

            int addDec = (plainTextDec + keyDec) % 26;

            char cipherChar = (char)(addDec + 'a');

            cipherStringSB.append(cipherChar);
        }

        return cipherStringSB.toString();
    }

    public static void main(String[] args) throws IOException {
        if (args.length != 2) {
            System.err.println("Usage: java vigenere keyFile plaintextFile");
            System.exit(1);
        }

        String keyArg = args[0];
        String plaintextArg = args[1];

        String keyString = readFile(keyArg);

        System.out.println();
        System.out.println();

        System.out.println("Vigenere Key:");
        System.out.println();

        printToConsole(keyString, 80);

        System.out.println();
        System.out.println();

        String plaintextStr = makingPlaintext(plaintextArg);

        System.out.println("Plaintext:");
        System.out.println();

        printToConsole(plaintextStr, 80);

        System.out.println();
        System.out.println();

        keyString = modifyKeyString(keyString);

        String cipherStr = buildingCiphertext(keyString, plaintextStr);

        System.out.println("Ciphertext:");
        System.out.println();

        printToConsole(cipherStr, 80);
    }
}

File: vigenere.cs
using System;
using System.IO;
using System.Text;
using System.Linq;

public class vigenere {

    private static string fileToString(string pathname) {
        string contents = File.ReadAllText(pathname);
        var sb = new StringBuilder();

        foreach (char i in contents)
            if (i != '\n' && i != '\r' && i != '\t')
                sb.Append(i);

        return new string(sb.ToString().Where(Char.IsLetter).ToArray()).ToLower();
    }

    public static void printToConsole(string sFile, int limit) {
        for (int i = 0; i < sFile.Length; i++) {
            Console.Write(sFile[i]);

            if ((i + 1) % limit == 0)
                Console.WriteLine();
        }
        Console.WriteLine();
    }

    public static string makingPlaintext(string plaintextArg) {
        var plaintextStr = new StringBuilder(fileToString(plaintextArg));

        while (plaintextStr.Length < 512)
            plaintextStr.Append('x');

        if (plaintextStr.Length > 512)
            return plaintextStr.ToString().Substring(0, 512);

        return plaintextStr.ToString();
    }

    public static string modifyKeyString(string keyString) {
        var keyStringSB = new StringBuilder(keyString);

        int initialKeyLen = keyStringSB.Length;
        int keyTracker = 0;

        while (keyStringSB.Length < 512) {
            // Notice the keyTracker post-increment
            keyStringSB.Append(keyStringSB[keyTracker++]);

            if (keyTracker >= initialKeyLen)
                keyTracker = 0;
        }
        return keyStringSB.ToString();
    }

    public static string buildingCiphertext(string keyString, string plaintextStr) {
        var cipherStringSB = new StringBuilder();

        for (int i = 0; i < plaintextStr.Length; i++) {
            int plainTextDec = plaintextStr[i] - 'a';
            int keyDec = keyString[i] - 'a';

            int addDec = (plainTextDec + keyDec) % 26;

            char cipherChar = (char)(addDec + 'a');

            cipherStringSB.Append(cipherChar);
        }
        return cipherStringSB.ToString();
    }

    public static void Main(string[] args) {
        if (args.Length != 2) {
            Console.Error.WriteLine("Usage: mono vigenere.exe keyFile plaintextFile");
            Environment.Exit(1);
        }

        string keyArg = args[0];
        string plaintextArg = args[1];

        string keyString = fileToString(keyArg);

        Console.WriteLine();
        Console.WriteLine();

        Console.WriteLine("Vigenere Key:");
        Console.WriteLine();

        printToConsole(keyString, 80);

        Console.WriteLine();
        Console.WriteLine();

        string plaintextStr = makingPlaintext(plaintextArg);

        Console.WriteLine("Plaintext:");
        Console.WriteLine();

        printToConsole(plaintextStr, 80);

        Console.WriteLine();
        Console.WriteLine();

        keyString = modifyKeyString(keyString);

        string cipherStr = buildingCiphertext(keyString, plaintextStr);

        Console.WriteLine("Ciphertext:");
        Console.WriteLine();

        printToConsole(cipherStr, 80);
    }
}
